File: src/main/java/com/logtechub/automation/reporting/AllureAttachments.java
package com.logtechub.automation.reporting;

import io.qameta.allure.Attachment;

public final class AllureAttachments {
    private AllureAttachments() {
    }

    @Attachment(value = "{name}", type = "text/plain")
    public static String attachText(String name, String content) {
        return content;
    }

    @Attachment(value = "{name}", type = "application/json")
    public static String attachJson(String name, String content) {
        return content;
    }

    @Attachment(value = "{name}", type = "image/png")
    public static byte[] attachPng(String name, byte[] content) {
        return content;
    }
}

File: src/automation/reporting/allure_attachments.py
from pathlib import Path

import allure


def attach_text(name: str, content: str) -> None:
    allure.attach(content, name=name, attachment_type=allure.attachment_type.TEXT)


def attach_png(name: str, content: bytes) -> None:
    allure.attach(content, name=name, attachment_type=allure.attachment_type.PNG)


def attach_file(path: str | Path, name: str | None = None) -> None:
    allure.attach.file(str(path), name=name or Path(path).name)
